src/utils/httpUtils.js

```diff
@@ -85,9 +86,50 @@ function doBulkUpsert(url, userToken, proxy, arr) {
       return resolve(res);
     });
   });
-}
+} // doBulkUpsert
+
+/**
+ * Find Refocus subjects using query parameters.
+ *
+ * @param  {String} url - The Refocus url.
+ * @param  {String} token - The Authorization token to use.
+ * @param  {String} proxy - Optional proxy url
+ * @param {String} qry - the query string
+ * @returns {Promise} array of subjects matching the query
+ */
+function findSubjects(url, token, proxy, qry) {
+  debug('findSubjects(url=%s, token=%s, proxy=%s, qry=%s)', url,
+    token ? 'HAS_TOKEN' : 'MISSING', proxy, qry);
+  if (!url) {
+    const e = new ValidationError('Missing refocus url');
+    logger.error('findSubjects', e.message);
+    return Promise.reject(e);
+  }
+
+  if (!qry) {
```

Comment:
subjectQuery is not a required field on the generator model. If we make it required, do we still have to pass in a queryParam to get all the subjects ?

Reply:
When we were taking a list of subjects or a subject query, one or the other was required; once we update the model in Refocus db and api to get rid of the array of subjects, the subject query field should be required. If there is ever a legit use case for generating samples for *all* subjects, then they can use something like "?absolutePath=*", but I'd rather enforce that people think about selecting the "right" subjects